u_roller.go

```diff
@@ -0,0 +1,110 @@
+package tls
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"net"
+	"sync"
+	"time"
+)
+
+type Roller struct {
+	HelloIDs            []ClientHelloID
+	HelloIDMu           sync.Mutex
+	NextHelloID         ClientHelloID
+	TcpDialTimeout      time.Duration
+	TlsHandshakeTimeout time.Duration
+}
+
+// NewRoller creates Roller object with default range of HelloIDs to cycle through until a
+// working/unblocked one is found.
+func NewRoller() (*Roller, error) {
+	tcpDialTimeoutInc, err := getRandUint16()
+	if err != nil {
+		return nil, err
+	}
+	tcpDialTimeoutInc = tcpDialTimeoutInc % 14
```

Comment:
I think this would be clearer as
```go
tcpDialTimeout := rand % 14 + 7
```
and similarly for other random constants.

Reply:
Good point.